### PR TITLE
Grmhd spherical blast wave analytic data

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Generalizes the cylindrical blast wave analytic data to be able to choose either a cylindrical or a spherical explosion

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Executables/GrMhd/ValenciaDivClean/EvolveValenciaDivCleanFwd.hpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ class SmoothFlow;
 }  // namespace Solutions
 namespace AnalyticData {
 class BondiHoyleAccretion;
-class CylindricalBlastWave;
+class BlastWave;
```
Comment (fmahebert) [inline src/Evolution/Executables/GrMhd/ValenciaDivClean/EvolveValenciaDivCleanFwd.hpp:29]: Alphabetize perhaps?

File: src/PointwiseFunctions/AnalyticData/GrMhd/CMakeLists.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@ spectre_target_sources(
   ${LIBRARY}
   PRIVATE
   BondiHoyleAccretion.cpp
-  CylindricalBlastWave.cpp
+  BlastWave.cpp
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/CMakeLists.txt:11]: Alphabetize (and headers below)

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/CMakeLists.txt [hunk 3]
```diff
@@ -5,7 +5,7 @@ set(LIBRARY "Test_GrMhdAnalyticData")
 
 set(LIBRARY_SOURCES
   Test_BondiHoyleAccretion.cpp
-  Test_CylindricalBlastWave.cpp
+  Test_BlastWave.cpp
```
Comment (fmahebert) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/CMakeLists.txt:7]: Alphabetize

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp [hunk 4]
```diff
@@ -149,7 +171,7 @@ void test_density_on_and_near_boundaries() {
 
   BlastWaveProxy cylindrical_blast_wave(
       inner_radius, outer_radius, inner_density, outer_density, inner_pressure,
-      outer_pressure, magnetic_field, adiabatic_index);
+      outer_pressure, magnetic_field, adiabatic_index, geometry);
```
Comment (fmahebert) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp]: if `geometry` is specifiable, the variable should be renamed `cylindrical_blast_wave` -> `blast_wave`

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp [hunk 5]
```diff
@@ -106,13 +124,16 @@ void test_variables(const DataType& used_for_size) noexcept {
...
       inner_radius, outer_radius, inner_density, outer_density, inner_pressure,
-      outer_pressure, magnetic_field, adiabatic_index);
+      outer_pressure, magnetic_field, adiabatic_index,
+      geometry == grmhd::AnalyticData::BlastWave::Geometry::Cylindrical ? 1.0
+                                                                        : 0.0);
 
   BlastWaveProxy cylindrical_blast_wave(
       inner_radius, outer_radius, inner_density, outer_density, inner_pressure,
-      outer_pressure, magnetic_field, adiabatic_index);
+      outer_pressure, magnetic_field, adiabatic_index, geometry);
```
Comment (fmahebert) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp]: if `geometry` is specifiable, the variable should be renamed `cylindrical_blast_wave` -> `blast_wave`

File: src/PointwiseFunctions/AnalyticData/GrMhd/BlastWave.cpp [hunk 6]
```diff
@@ -18,32 +20,34 @@
...
-Scalar<DataType> compute_piecewise(const tnsr::I<DataType, 3>& x,
-                                   const double inner_radius,
-                                   const double outer_radius,
-                                   const double inner_value,
-                                   const double outer_value) noexcept {
-  const DataType cylindrical_radius =
-      sqrt(square(get<0>(x)) + square(get<1>(x)));
+Scalar<DataType> compute_piecewise(
+    const tnsr::I<DataType, 3>& x,
+    const grmhd::AnalyticData::BlastWave::Geometry geometry,
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/BlastWave.cpp]: [super nitpick, feel free to veto] Put the Geometry argument last to match all other functions

**Review by fmahebert — COMMENTED** [review]
A nice generalization 👍 

Edit to add: you can squash changes directly

**Comment by nilsdeppe** [comment]
I've pushed fixups. Thanks for reviewing :)

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp (hunk 4 above)
Comment (fmahebert) [reply tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp]: ping on this one

**Review by fmahebert — COMMENTED** [review]
LGTM, I think I see just one comment not yet addressed. Please squash including this last one

**Comment by nilsdeppe** [comment]
Done and squashed. Thanks for reviewing :)

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp [hunk 7]
```diff
@@ -147,9 +167,9 @@ void test_density_on_and_near_boundaries() {
   const std::array<double, 3> magnetic_field{{0.1, 0.0, 0.0}};
   const double adiabatic_index = 1.3333333333333333333;
 
-  CylindricalBlastWaveProxy cylindrical_blast_wave(
-      inner_radius, outer_radius, inner_density, outer_density, inner_pressure,
-      outer_pressure, magnetic_field, adiabatic_index);
+  BlastWaveProxy cal_blast_wave(inner_radius, outer_radius, inner_density,
```
Comment (fmahebert) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_BlastWave.cpp]: maybe the third time will be the charm 🙃

File: docs/References.bib [hunk 8]
```diff
@@ -297,6 +297,40 @@ @article{Casoni2012
   year    = 2013
 }
 
+@article{CerdaDuran2007,
+  author =       {{Cerd{\'a}-Dur{\'a}n}, P. and {Font}, J.~A. and {Dimmelmeier},
+                  H.},
+  title =        "General relativistic simulations of
+                  passive-magneto-rotational core collapse with microphysics",
+  journal =      {\aap},
```
Comment (kidder) [inline docs/References.bib]: fix (either supply the journal or add whatever file is missing that defines `\aap`

File: docs/References.bib [hunk 9]
```diff
@@ -297,6 +297,40 @@ @article{Casoni2012
...
+  archivePrefix ={arXiv},
+  eprint =       {astro-ph/0703597}
+}
+
+@article{CerdaDuran2008,
+  author =       {{Cerd{\'a}-Dur{\'a}n}, P. and {Font}, J.~A. and {Ant{\'o}n},
+                  L. and {M{\"u}ller}, E.},
+  title =        "A new general relativistic magnetohydrodynamics code for
+                  dynamical spacetimes",
+  journal =      {\aap},
```
Comment (kidder) [inline docs/References.bib]: fix

File: docs/References.bib [hunk 10]
```diff
@@ -316,6 +350,23 @@ @article{CHEN2016604
...
+  title =        "{Spritz: a new fully general-relativistic magnetohydrodynamic
+                  code}",
+  journal =      {Classical and Quantum Gravity},
+  year =         2020,
+  month =        jul,
+  volume =       37,
+  number =       13,
+  eid =          135010,
+  pages =        135010,
+  doi =          {10.1088/1361-6382/ab8be8},
```
Comment (kidder) [inline docs/References.bib:365]: this needs the url added

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
I have tried again :D Thanks for reviewing :)

**Review by kidder — APPROVED** [review]

**Pull request merged**